File: Data Structures & Algorithms/majority-element/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        let len = nums.length;
        const map = new Map();
        for (const num of nums) {
            map.set(num, (map.get(num) || 0) + 1)
            if(map.get(num)>len/2) return num
        }
    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        dp = {0}
        total = sum(nums)
        if total%2:
            return False
        target = total//2
        for i in range(len(nums)):
            nextDp = set()
            for num in dp:
                if num+nums[i]==target:
                    return True
                nextDp.add(num+nums[i])
            dp.update(nextDp)
        return True if target in dp else False